ducktape/tests/test.py

```diff
@@ -137,3 +129,168 @@ def is_test_class(self, obj):
         return inspect.isclass(obj) and issubclass(obj, Test) and len(obj.__subclasses__()) == 0
 
 
+class TestReporter(object):
+    def __init__(self, results):
+        self.results = results
+
+    def pass_fail(self, success):
+        return "PASS" if success else "FAIL"
+
+    def report(self):
+        raise NotImplementedError("method report must be implemented by subclasses of TestReporter")
+
+
+class SimpleReporter(TestReporter):
+    def header_string(self):
+        header = ""
+        header += "Test run with session_id " + self.results.test_session_context.test_session_id + "\n"
+        header += self.pass_fail(self.results.get_aggregate_success()) + "\n"
+        header += "------------------------------------------------------------\n"
+
+        return header
+
+    def result_string(self, result):
```

Comment:
This probably doesn't matter much yet, but you probably want the lower level interface to be some sort of write-to-stream interface (i.e. write to a file) so you're not just collecting a ton of string data in memory. If you end up needing to collect it all in memory you can use something like StringIO to provide a file-like interface from which you can then grab the full output.

At a minimum, you probably want to change these to collect all the substrings in an array and then do a join. Concatenating like this gets expensive quickly since all the data is copied every time the string is extended.